### PR TITLE
add `fireXxxxEventAsync` methods to MEvent;

### LINKED ISSUE #9981: asynchronous ireEventAsync, fireDataEventAsync, and fireNonBubblingEventAsync do not exists
The documented methods fireEventAsync, fireDataEventAsync, and fireNonBubblingEventAsync do not exists im MEvent.

Only in qx.event.Registration you found     fireEventAsync : function(target, type, clazz, args).

Should we remove the doc or should we implement the methods?

### PR DESCRIPTION
**Pull request by johnspackman**
fixes https://github.com/qooxdoo/qooxdoo/issues/9981

### CODE REVIEW

**Review by cboulanger — APPROVED** [review]

**Review by hkollmann — APPROVED** [review]

**Review by oetiker — APPROVED** [review]

File: framework/source/class/qx/core/MEvent.js [hunk 1]
```diff
@@ -186,6 +186,31 @@ qx.Mixin.define("qx.core.MEvent",
...
+    {
+      if (!qx.core.Environment.get("qx.promise")) {
+        throw new Error(this.classname + ".fireEventAsync not supported because qx.promise==false");
+      }
+      
+      if (!this.$$disposed) {
+        return this.__Registration.fireEventAsync(this, type, clazz, args);
+      }
+
+      return qx.Promise.resolve(true);
```
Comment (derrell) [inline framework/source/class/qx/core/MEvent.js:210]: Why does this resolve if `this` is disposed? I'd think it should reject. It's a programmer error to fire an event on a disposed object...

File: framework/source/class/qx/core/MEvent.js [hunk 2]
```diff
@@ -208,6 +233,34 @@ qx.Mixin.define("qx.core.MEvent",
...
+    {
+      if (!qx.core.Environment.get("qx.promise")) {
+        throw new Error(this.classname + ".fireNonBubblingEventAsync not supported because qx.promise==false");
+      }
+      
+      if (!this.$$disposed) {
+        return this.__Registration.fireNonBubblingEventAsync(this, type, clazz, args);
+      }
+
+      return qx.Promise.resolve(true);
```
Comment (derrell) [inline framework/source/class/qx/core/MEvent.js:260]: ditto

File: framework/source/class/qx/core/MEvent.js [hunk 3]
```diff
@@ -236,6 +289,42 @@ qx.Mixin.define("qx.core.MEvent",
...
+      {
+        if (oldData === undefined) {
+          oldData = null;
+        }
+        return this.__Registration.fireEventAsync(
+          this, type, qx.event.type.Data, [data, oldData, !!cancelable]
+        );
+      }
+
+      return qx.Promise.resolve(true);
```
Comment (derrell) [inline framework/source/class/qx/core/MEvent.js:327]: ditto

**Review by derrell — CHANGES_REQUESTED** [review]

File: framework/source/class/qx/core/MEvent.js (hunk 1 above)
Comment (johnspackman) [reply framework/source/class/qx/core/MEvent.js:210]: I agree, but that's how the ordinary, non-async `fireEvent` works so this is mirroring functionality but just wrapping it in a promise.

Comment (derrell) [reply framework/source/class/qx/core/MEvent.js:210]: Sigh.

**Review by derrell — APPROVED** [review]

**Review by level420 — APPROVED** [review]

**Pull request merged**